sympy/core/expr.py

```diff
@@ -362,11 +379,16 @@ def __lt__(self, other):
         n2 = _n2(self, other)
         if n2 is not None:
             return _sympify(n2 < 0)
-        if self.is_extended_real or other.is_extended_real:
-            dif = self - other
-            if dif.is_extended_negative is not None and \
-                    dif.is_extended_negative is not dif.is_extended_nonnegative:
-                return sympify(dif.is_extended_negative)
+        if self.is_extended_real and other.is_extended_real:
+            if self.is_infinite and other.is_infinite:
+                if (self * other).is_extended_positive:  # inf < inf & -inf < -inf
+                    return S.false
```

Comment:
It still looks like this can be simplified. Are these three lines needed given the lines below?

Reply:
Looks like I somehow looked at an out of date version...